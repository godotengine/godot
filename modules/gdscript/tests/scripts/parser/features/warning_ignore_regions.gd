@warning_ignore_start("unreachable_code", "narrowing_conversion")

var _a = 1
@warning_ignore_start("unused_private_class_variable")
var _b = 2
var _c = 3
@warning_ignore_restore("unused_private_class_variable")
var _d = 4

func test():
	return

	var a = 1
	@warning_ignore_start("unused_variable")
	var b = 2
	var c = 3
	@warning_ignore_restore("unused_variable")
	var d = 4

	var _x: int = 1.0
	@warning_ignore_restore("narrowing_conversion")
	var _y: int = 1.0

func test_2():
	return
	print(42)
