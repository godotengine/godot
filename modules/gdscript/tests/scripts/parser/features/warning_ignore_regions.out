GDTEST_OK
~~ WARNING at line 3: (UNUSED_PRIVATE_CLASS_VARIABLE) The class variable "_a" is declared but never used in the class.
~~ WARNING at line 8: (UNUSED_PRIVATE_CLASS_VARIABLE) The class variable "_d" is declared but never used in the class.
~~ WARNING at line 13: (UNUSED_VARIABLE) The local variable "a" is declared but never used in the block. If this is intended, prefix it with an underscore: "_a".
~~ WARNING at line 18: (UNUSED_VARIABLE) The local variable "d" is declared but never used in the block. If this is intended, prefix it with an underscore: "_d".
~~ WARNING at line 22: (NARROWING_CONVERSION) Narrowing conversion (float is converted to int and loses precision).
